### PR TITLE
[PM-12252] Autofill name of new ciphers

### PR DESCRIPTION
**Pull request by nick-livefront**
## 🎟️ Tracking

[PM-12252](https://bitwarden.atlassian.net/browse/PM-12252)

## 📔 Objective

Autofill the `name` when creating a new cipher on the extension refresh components
- Luckily this was already built in, just needed to be populated!

## 📸 Screenshots

https://github.com/user-attachments/assets/b7cffaa3-e3ac-4c01-8a24-07a85d691375

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-12252]: https://bitwarden.atlassian.net/browse/PM-12252?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shane-melton — COMMENTED** [review]
This looks good, but I do have a few questions before we merge:

- Should this be for all cipher types or just `Login`?
- If the extension is popped out, should we ignore the current tab? (This looks to be the case with the current extension)

cc: @danielleflinn

**Comment by danielleflinn** [comment]
@shane-melton good catch on the type edge case, yes we should only add the default item name for Login item types.

And yes, lets match the popout's behavior to what the existing extension does.

**Comment by nick-livefront** [comment]
Thanks, @shane-melton and @danielleflinn! Updated as such  [01006d3](https://github.com/bitwarden/clients/pull/11127/commits/01006d32b691a1bd96e5ecf7eb65655f709826e7) & [249c655](https://github.com/bitwarden/clients/pull/11127/commits/249c655c4fa43b74338e0fa65c0811faeb2bd73c)

File: apps/browser/src/vault/popup/components/vault-v2/new-item-dropdown/new-item-dropdown-v2.component.ts [hunk 1]
```diff
@@ -37,18 +40,30 @@ export class NewItemDropdownV2Component {
     private dialogService: DialogService,
   ) {}
 
-  private buildQueryParams(type: CipherType): AddEditQueryParams {
+  private async buildQueryParams(type: CipherType): Promise<AddEditQueryParams> {
+    const tab = await BrowserApi.getTabFromCurrentWindow();
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/new-item-dropdown/new-item-dropdown-v2.component.ts:44]: 🎨   There's a small chance the `initialValues.uri` could get out of sync with the `tab.url` here. Looking at the `vault-v2` component, we [set `initialValues.uri` using the `getTabFromCurrentWindow()`](https://github.com/bitwarden/clients/blob/ade01c9d078613f6ba176209192238ffe59ee684/apps/browser/src/vault/popup/components/vault/vault-v2.component.ts#L75) which could be a different tab by the time we call `buildQueryParams()`.

I'm thinking we should remove `uri` from the `initialValues` object and instead set `uri` and `name` values using the `tab` you're fetching here if the extension is not in a popout. How does that sound to you?

File: apps/browser/src/vault/popup/components/vault-v2/new-item-dropdown/new-item-dropdown-v2.component.ts [hunk 2]
```diff
@@ -37,18 +39,30 @@ export class NewItemDropdownV2Component {
...
+    const poppedOut = BrowserPopupUtils.inPopout(window);
+
+    const loginDetails: { uri?: string; name?: string } = {};
+
+    // When a Login Cipher is created and the extension is not popped out,
+    // pass along the uri and name
+    if (!poppedOut && type === CipherType.Login) {
+      loginDetails.uri = tab?.url;
+      loginDetails.name = Utils.getHostname(tab.url);
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/new-item-dropdown/new-item-dropdown-v2.component.ts:52]: ⛏️ `tab?.url` here as well. Or perhaps we check if the `tab` is null in the `if` condition?

**Review by shane-melton — CHANGES_REQUESTED** [review]
Oh, looks like we need to fix a few tests then we should be good to go!

**Comment by nick-livefront** [comment]
> Oh, looks like we need to fix a few tests then we should be good to go!

Moving a little too quick for my own good

**Review by shane-melton — APPROVED** [review]

**Pull request merged**